### ISSUE TITLE
false PossiblyUnusedParam errors and eval()

### ISSUE THREAD

**Issue by terrafrost** [vimeo/psalm#11930]
```php
<?php

class Demo
{
    public function run(): void
    {
        $this->test(['k' => '1']);
    }

    protected function test(array $in): void
    {
        echo $in['k'];
        eval('$x = 1;');
    }
}

$repro = new Demo();
$repro->run();
```
Running psalm on that gives this error:

```
ERROR: PossiblyUnusedParam
at /path/to/demo.php:10:35
Param #1 is never referenced in this method (see https://psalm.dev/134)
    protected function test(array $in): void
```
Interestingly, changing the visibility of the test method from protected to private makes the error go away. 

The location of the `eval()` also doesn't matter. It could be before or after the echo and you still get it.
